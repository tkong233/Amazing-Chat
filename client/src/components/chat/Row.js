import React from 'react';
import { connect } from 'react-redux';
import Message from './Message';

const LOADING = 1;
const LOADED = 2;

const Row = (props) => {
  const { index, style } = props;
  const { itemStatusMap, messages, sender } = props.chat;
  let label;
  console.log('itemStatusMap', itemStatusMap);
  if (itemStatusMap[index] === LOADED) {
    // label = `Row ${index}`;
    const message = messages[index];
    label = (<Message
              text={message.message}
              sender={message.from}
              user={sender}
              from={message.from}
              to={message.to}
              message={message.message}
              datetime={message.datetime}
              pairId={message.pairId}
            />)
  } else if (itemStatusMap[index] === LOADING) {
    // console.log(index, itemStatusMap[index]);
    label = "Loading...";
  }
  return (
    <div className="ListItem" style={style}>
      {label}
    </div>
  );
}

const mapStateToProps = state => ({
  chat: state.chat,
})

export default connect(
  mapStateToProps,
  {}
)(Row);