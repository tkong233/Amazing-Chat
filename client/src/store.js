import { createStore, applyMiddleware, compose } from "redux";
import thunk from "redux-thunk";
import { persistStore, persistReducer } from 'redux-persist'
import rootReducer from "./reducers";
import storage from 'redux-persist/lib/storage' // defaults to localStorage for web

const initialState = {};

const persistConfig = {
  key: 'root',
  storage,
}

const middleware = [thunk];

const persistedReducer = persistReducer(persistConfig, rootReducer, initialState);

const store = createStore(

  persistedReducer,
  compose(
    applyMiddleware(...middleware),
    (window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ &&
      window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__()) ||
      compose
  )
);

const persistor = persistStore(store);

export {
  store,
  persistor,
  middleware
};



