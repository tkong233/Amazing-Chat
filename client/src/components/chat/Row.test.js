import React from 'react';
import Row from './Row';
import {shallow} from 'enzyme';
import checkPropTypes from 'check-prop-types';
import Enzyme from 'enzyme';
import EnzymeAdapter from 'enzyme-adapter-react-16'; 
import { applyMiddleware, createStore } from 'redux';
import {middleware} from '../../store';
import rootReducer from '../../reducers/index'

Enzyme.configure({
    adapter: new EnzymeAdapter(),
});

const setUp = (initialState={}) =>{
    const createStoreWithMiddleware = applyMiddleware(...middleware)(createStore);
    const store = createStoreWithMiddleware(rootReducer, initialState);
    const wrapper = shallow(<Row store={store} />).dive();
    // console.log(wrapper.debug());
    return wrapper;
}

describe('Row Component', ()=>{
    describe('Checking PropTypes', ()=>{
        it('Should not throw a warning', ()=>{
            const expectedProps = {
                chat: {},
            }
            // eslint-disable-next-line react/forbid-foreign-prop-types
            const propsErr = checkPropTypes(Row.propTypes, expectedProps, 'props', Row.name);
            expect(propsErr).toBeUndefined();
        })
    });
    describe('Renders', ()=>{
        let wrapper;
        beforeEach(()=>{
            const initialState = {
                chat:{
                    itemStatusMap:
                    { 1 : 'LOADED'},
                    messages:[{
                        pairId: '',
                        from: 'userA', 
                        to: 'userB',
                        message: 'test message',
                        datetime: '2020/09/01',
                        type: 'text'
                    }],
                    sender: 'userA',
                    receiver: 'userB'
                }
            };
            wrapper = setUp(initialState);
        });

        it('Should render without errors', ()=>{
            const container = wrapper.find(`[data-test='RowComponent']`);
            expect(container.length).toBe(1);
        });

        it('Snapshot testing', ()=>{
            expect(wrapper).toMatchSnapshot();
        });


    });
});