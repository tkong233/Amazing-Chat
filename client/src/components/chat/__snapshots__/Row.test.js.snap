// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row Component Renders Snapshot testing 1`] = `ShallowWrapper {}`;
